src/Configuration/Config.Json/src/JsonConfigurationFileParser.cs

```diff
@@ -19,89 +17,70 @@ internal class JsonConfigurationFileParser
         {
             _data.Clear();
-            _reader = new JsonTextReader(new StreamReader(input));
-            _reader.DateParseHandling = DateParseHandling.None;
-
-            var jsonConfig = JObject.Load(_reader);
 
-            VisitJObject(jsonConfig);
+            using (var reader = new StreamReader(input))
+            using (JsonDocument doc = JsonDocument.Parse(reader.ReadToEnd(),
+                new JsonReaderOptions { CommentHandling = JsonCommentHandling.Skip }))
+            {
+                VisitElement(doc.RootElement);
+            }
 
             return _data;
         }
 
-        private void VisitJObject(JObject jObject)
-        {
-            foreach (var property in jObject.Properties())
+        private void VisitElement(JsonElement element) {
+            foreach (var property in element.EnumerateObject())
             {
                 EnterContext(property.Name);
-                VisitProperty(property);
+                VisitValue(property.Value);
                 ExitContext();
             }
         }
 
-        private void VisitProperty(JProperty property)
+        private void VisitValue(JsonElement value)
         {
-            VisitToken(property.Value);
-        }
-
-        private void VisitToken(JToken token)
-        {
-            switch (token.Type)
-            {
-                case JTokenType.Object:
-                    VisitJObject(token.Value<JObject>());
+            switch (value.Type) {
+                case JsonValueType.Object:
+                    VisitElement(value);
                     break;
 
-                case JTokenType.Array:
-                    VisitArray(token.Value<JArray>());
+                case JsonValueType.Array:
+                    var index = 0;
+                    foreach (var arrayElement in value.EnumerateArray()) {
+                        EnterContext(index.ToString());
+                        VisitValue(arrayElement);
+                        ExitContext();
+                        index++;
+                    }
                     break;
 
-                case JTokenType.Integer:
-                case JTokenType.Float:
-                case JTokenType.String:
-                case JTokenType.Boolean:
-                case JTokenType.Bytes:
-                case JTokenType.Raw:
-                case JTokenType.Null:
-                    VisitPrimitive(token.Value<JValue>());
+                case JsonValueType.Number:
+                case JsonValueType.String:
+                case JsonValueType.True:
+                case JsonValueType.False:
+                case JsonValueType.Null:
+                    var key = _currentPath;
+                    if (_data.ContainsKey(key))
+                    {
+                        throw new FormatException(Resources.FormatError_KeyIsDuplicated(key));
+                    }
+                    _data[key] = value.ToString(CultureInfo.InvariantCulture);
```

Comment:
FYI @ajcvickers there is no overload that takes culture info in the new stack currently